modules/kmm-replacing-in-tree-modules-with-out-of-tree-modules.adoc

```diff
@@ -0,0 +1,41 @@
+// Module included in the following assemblies:
+//
+// * hardware_enablement/kmm-kernel-module-management.adoc
+
+:_content-type: CONCEPT
+[id="kmm-replacing-in-tree-modules-with-out-of-tree-modules_{context}"]
+= Replacing in-tree modules with out-of-tree modules
+
+You can use Kernel Module Management (KMM) to build kernel modules that can be loaded or unloaded into the kernel on demand. These modules extend the functionality of the kernel without the need to reboot the system. Modules can be configured as built-in or dynamically loaded.
+
+Dynamically loaded modules include in-tree modules and out-of-tree (OOT) modules. In-tree modules are internal to the Linux kernel tree, that is, they are already part of the kernel. Out-of-tree modules are external to the Linux kernel tree. They are generally written for development and testing purposes, such as testing the new version of a kernel module that is shipped in-tree, or to deal with incompatibilities.
+
+Some modules loaded by KMM could replace in-tree modules already loaded on the node. To unload an in-tree module before loading your module, set the `.spec.moduleLoader.container.inTreeModuleToRemove`. This is an example for module replacement for all kernel mappings:
+
+[source,yaml]
+----
+spec:
+  moduleLoader:
+    container:
+      modprobe:
+        moduleName: mod_a
+
+      inTreeModuleToRemove: mod_b
+----
+
+In this example, the ModuleLoader pod uses `inTreeModuleToRemove` to unload the in-tree `mod_b` before loading `mod_a`
+from the ModuleLoader image.
+When the ModuleLoader pod is terminated and `mod_a` is unloaded, `mod_b` will not be loaded again.
+
+This is an example for module replacement for specific kernel mappings:
+
+[source,yaml]
+----
+spec:
+    moduleLoader:
```

Comment:
Can we use the same layout as in the above example ? I see 2 spaces in the previous example and in here I see 4 spaces. 
It would be nice to have the same spacing.

Reply:
@cdvultur changed.